sample-geovelocity/siddhi-execution-geovelocity/src/main/java/org/wso2/extension/siddhi/execution/geovelocity/internal/impl/RDBMSGeoVelocityDataResolver.java

```diff
@@ -0,0 +1,165 @@
+import java.sql.SQLException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This is the implementation class that provides the RDBMS based approach to get geovelocity related data.
+ */
+public class RDBMSGeoVelocityDataResolver {
+
+    private static final Log log = LogFactory.getLog(RDBMSGeoVelocityDataResolver.class);
+    private static final RDBMSGeoVelocityDataResolver instance = new RDBMSGeoVelocityDataResolver();
+    private static final String CONFIG_KEY_ISPERSIST_IN_DATABASE = "isPersistInDatabase";
+    private static final String CONFIG_KEY_DATASOURCE = "datasource";
+    private static final String DEFAULT_DATASOURCE_NAME = "IS_ANALYTICS_DB";
+    private AtomicBoolean isInitialized = new AtomicBoolean(false);
+    private DatabaseUtils dbUtils;
+    private boolean isPersistInDatabase;
+
+    private static final String SQL_SELECT_LASTLOGINTIME_FROM_GEOVELOCITY_INFO = "SELECT logintime FROM " +
+            "Geovelocity_info WHERE username = ? AND city = ? AND authenticationsuccess = 1";
+    private static final String SQL_SELECT_ID_FROM_TRAVELRESTRICTEDAREAS = "SELECT count(ID) FROM " +
+            "travelrestrictedareas WHERE (currentlocation = ? AND lastlocation = ?)" +
+            "OR (currentlocation = ? AND lastlocation = ?)";
+
+    public static RDBMSGeoVelocityDataResolver getInstance() {
+        return instance;
+    }
+
+    public void init(ConfigReader configReader) throws GeoVelocityException {
+
+        if (isInitialized.get()) {
+
+            return;
+        }
+        isPersistInDatabase = Boolean.parseBoolean(configReader.readConfig(CONFIG_KEY_ISPERSIST_IN_DATABASE, "true"));
+        dbUtils = DatabaseUtils.getInstance();
+        dbUtils.initialize(configReader.readConfig(CONFIG_KEY_DATASOURCE, DEFAULT_DATASOURCE_NAME));
+        isInitialized.set(true);
+    }
+
+    public GeoVelocityData getGeoVelocityData(String username, String city) {
+
+        GeoVelocityData geoVelocityData = null;
+        Connection connection = null;
+        try {
+            connection = dbUtils.getConnection();
+            geoVelocityData = loadGeoVelocityData(username, city, connection);
+        } catch (SQLException e) {
+            log.error("Cannot retrieve the login time of the last " +
+                    "successful login from the give location from database ", e);
+        } finally {
+            dbUtils.closeAllConnections(null, connection, null);
+        }
+        return geoVelocityData;
+    }
+
+    public GeoVelocityData getRestrictedLocations(String currentCity, String previousCity,
+                                                  String currentCountry, String previousCountry) {
+
+        GeoVelocityData geoVelocityData = null;
+        Connection connection = null;
+        try {
+            connection = dbUtils.getConnection();
+            geoVelocityData = loadLoginData(currentCity, previousCity,
+                    currentCountry, previousCountry, connection);
```

Comment:
These two lines can be one line

Reply:
Fixed in https://github.com/wso2/samples-is/pull/265